product_customer_code/product.py

```diff
@@ -44,30 +41,23 @@ def copy(self, default=None):
         res = super(ProductProduct, self).copy(default=default)
         return res
 
-    def name_search(self, cr, user, name='', args=None, operator='ilike',
-                    context=None, limit=80):
-        res = super(ProductProduct, self).name_search(
-            cr, user, name, args, operator, context, limit)
-        if not context:
-            context = {}
-        product_customer_code_obj = self.pool.get('product.customer.code')
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=80):
+        if args is None:
+            args = []
+        res = super(ProductProduct, self).name_search(name=name, args=args,
+                                                      operator=operator,
+                                                      limit=limit)
+        product_customer_code_obj = self.env['product.customer.code']
         if not res:
-            ids = []
-            partner_id = context.get('partner_id', False)
+            partner_id = self._context.get('partner_id')
             if partner_id:
-                id_prod_code = \
-                    product_customer_code_obj.search(cr, user,
-                                                     [('product_code',
-                                                       '=', name),
-                                                         ('partner_id', '=',
-                                                          partner_id)],
-                                                     limit=limit,
-                                                     context=context)
-                # TODO: Search for product customer name
-                id_prod = id_prod_code and product_customer_code_obj.browse(
-                    cr, user, id_prod_code, context=context) or []
-                for ppu in id_prod:
-                    ids.append(ppu.product_id.id)
-            if ids:
-                res = self.name_get(cr, user, ids, context)
+                id_prod_code = product_customer_code_obj.search(
+                    [('product_code', '=', name),
+                     ('partner_id', '=', partner_id)], limit=limit,)
+                id_prod = id_prod_code and (
+                    product_customer_code_obj.browse(
+                        id_prod_code, context=self._context) or [])
+            if id_prod:
+                res = id_prod.name_get()
```

Comment:
You don't need `name_get` of `product_customer_code_obj` you need `name_get` of `product.product`